examples/step-44/step-44.cc

```diff
@@ -2001,16 +1998,18 @@ namespace Step44
       {
         fe_values_ref.reinit(cell);
 
-        PointHistory<dim> *lqph =
-          reinterpret_cast<PointHistory<dim>*>(cell->user_pointer());
-
-        Assert(lqph >= &quadrature_point_history.front(), ExcInternalError());
-        Assert(lqph <= &quadrature_point_history.back(), ExcInternalError());
+        // In contrast to that which was previously called for,
+        // in this instance the quadrature point data is specifically
+        // non-modifiable since we will only be accessing data.
+        const std::vector<std::shared_ptr<const PointHistory<dim> > > lqph =
+            const_cast<const CellDataStorage<typename Triangulation<dim>::cell_iterator,
+                                             PointHistory<dim> >&>(quadrature_point_history).get_data(cell);
```

Comment:
That was the first thing I tried but, unfortunately, it doesn't work. `quadrature_point_history` is a non-constant member variable of the `Solid` class, and the calling function is also non-constant. So the compiler (`clang`, in this instance) seems to default to the non-constant `get_data` call and doesn't look for any suitable alternatives. Its not vital that I return pointers to`const PointHistory<dim>` objects, but I would prefer to do so since I think that its defensive programming.